crypto/src/merkle/proofs.rs

```diff
@@ -255,6 +255,168 @@ impl<H: Hasher> BatchMerkleProof<H> {
         v.remove(&1).ok_or(MerkleTreeError::InvalidProof)
     }
 
+    /// Computes the uncompressed Merkle paths which aggregate to this proof.
+    ///
+    /// # Errors
+    /// Returns an error if:
+    /// * No indexes were provided (i.e., `indexes` is an empty slice).
+    /// * Number of provided indexes is greater than 255.
+    pub fn into_paths(self, indexes: &[usize]) -> Result<Vec<Vec<H::Digest>>, MerkleTreeError> {
+        if indexes.is_empty() {
+            return Err(MerkleTreeError::TooFewLeafIndexes);
+        }
+        if indexes.len() > MAX_PATHS {
+            return Err(MerkleTreeError::TooManyLeafIndexes(
+                MAX_PATHS,
+                indexes.len(),
+            ));
+        }
+        let mut leaves = self.leaves.clone();
+        leaves.reverse();
+
+        let mut partial_tree_map = BTreeMap::new();
+
+        for i in indexes {
+            if let Some(leaf) = leaves.pop() {
+                partial_tree_map.insert(*i + (1 << (self.depth)), leaf);
+            } else {
+                return Err(MerkleTreeError::TooManyLeafIndexes(
+                    leaves.len(),
+                    indexes.len(),
+                ));
```

Comment:
Maybe this is something we can check just once before entering the loop? Also, it might be possible to simplify this loop as follows:
```Rust
for (&i, leaf) in indexes.iter().zip(leaves.iter()) {

}
```
This should also let us avoid cloning and reversing leaves above.

Reply:
Indeed!